src/actions/profile-view.js

```diff
@@ -807,12 +810,46 @@ export function hideProvidedTracks(
 
     for (const [pid, localTracksToHide] of localTracksByPidToHide) {
       const localTracks = getLocalTracks(getState(), pid);
+      const hiddenLocalTracks = getHiddenLocalTracks(getState(), pid);
       for (const trackIndex of localTracksToHide) {
         const localTrack = localTracks[trackIndex];
         if (localTrack.type === 'thread') {
           newSelectedThreadIndexes.delete(localTrack.threadIndex);
         }
       }
+
+      const newHiddenLocalTrackCount = new Set([
+        ...hiddenLocalTracks,
+        ...localTracksToHide,
+      ]).size;
+      if (newHiddenLocalTrackCount === localTracks.length) {
+        // Hiding these local tracks will hide all of the tracks for this process.
+        // At this point two different cases need to be handled:
+        //   1.) There is a main thread for the process, go ahead and hide all the
+        //       local tracks.
+        //   2.) There is no main thread for the process, attempt to hide the
+        //       processes' global track.
+        const { globalTrack, globalTrackIndex } = getGlobalTrackAndIndexByPid(
+          getState(),
+          pid
+        );
+
+        if (globalTrack.mainThreadIndex === null) {
+          // Since the process has no main thread, the entire process should be hidden.
+          dispatch(hideGlobalTrack(globalTrackIndex));
```

Comment:
It's not clear to me why you don't just add it to the set of `globalTracksToHide` (possibly we need to clone `globalTracksToHide` first so that we don't touch the passed value).